### ISSUE TITLE
Memory leak on BugsnagConfig

### ISSUE THREAD

**Issue by jbelkins** [bugsnag/bugsnag-cocoa#737]
### Description
Went looking for memory leaks in my own code today and ran across this on latest Bugsnag iOS SDK.

### Issue
Found that `BugsnagConfiguration` leaks due to a reference cycle between it and `BugsnagBreadcrumbs`.

The culprit is [here](https://github.com/bugsnag/bugsnag-cocoa/blob/master/Bugsnag/Configuration/BugsnagConfiguration.m#L223):
```
    _breadcrumbs = [[BugsnagBreadcrumbs alloc] initWithConfiguration:self];
```
`BugsnagBreadcrumbs` keeps a strong reference to the `self` you pass in, causing a reference cycle.  It's not major, but it's leaked memory.  How to fix is up to you.

### Environment
Library versions:
* Bugsnag version: 6.0.1 from Cocoapods

**Comment by xander-jones** [comment]
Hi @jbelkins, thanks for reporting this. We're going to be taking a look into this as a priority.

**Comment by johnkiely1** [comment]
This has now been fixed in [V6.1.1](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.1.1)
